### ISSUE TITLE
SA1119 is incorrectly reported for |= and &= statements

### ISSUE THREAD

**Issue by 0x084E** [DotNetAnalyzers/StyleCopAnalyzers#1302]
The following code raises SA1119 but the parenthesis are necessary:

```csharp
bool value = false;
value |= (true && true);
```

**Comment by sharwell** [comment]
Hi @0x084E and welcome to the project!

I have a quick question to help clarify this:

:question: Why are the parentheses necessary?

**Comment by 0x084E** [comment]
They are not, my fault...
[RefactoringEssentials](https://github.com/icsharpcode/RefactoringEssentials) did this during a refactoring. For some reason I felt certain that this was necessary and didn't check it.
Sorry, next time I'll check my assertions ;).

**Comment by sharwell** [comment]
@0x084E Don't worry about it, we all miss these things from time to time. :smile:
